docs/new-project-setup.md

```diff
@@ -0,0 +1,310 @@
+At best this makes the tools unusable, and at worst it can lock your Snowflake account.
+One example of such a tool is (as of this writing) the Snowflake Terraform Provider.
+
+The recommended workaround for this is to add a key pair to your account for use with those tools.
+
+1. Follow the instructions given [here](https://docs.snowflake.com/en/user-guide/key-pair-auth#configuring-key-pair-authentication)
+    to generate a key pair and add the public key to your account.
+    Keep the key pair in a secure place on your device.
+    [This gist](https://gist.github.com/ian-r-rose/1c714ee04be53f7a3fd80322e1a22c27)
+    contains the bash commands from the instructions,
+    and can be helpful for quickly creating a new encrypted key pair.
+    Usage of the script looks like:
+    ```bash
+    bash generate_encrypted_key.sh <key-name> <passphrase>
+    ```
+    You can use `pbcopy < _your_public_key_file_name_.pub` to copy the contents of your public key.
+    Be sure to remove the `----BEGIN PUBLIC KEY----` and `-----END PUBLIC KEY------` portions
+    when adding your key to your Snowflake user.
+1. In your local `.bash_profile` or an `.env` file, add environment variables for
+    `SNOWFLAKE_ACCOUNT`, `SNOWFLAKE_USER`, `SNOWFLAKE_PRIVATE_KEY_PATH`,
+    and (if applicable) `SNOWFLAKE_PRIVATE_KEY_PASSPHRASE`.
+
+### Apply a session policy
+
+By default, Snowflake logs out user sessions after four hours of inactivity.
+ODI's information security policies prefer that we log out after one hour of inactivity for most accounts,
+and after fifteen minutes of inactivity for particularly sensitive accounts.
+
+!!! note
+    It's possible we will do this using Terraform in the future,
+    but at the time of this writing the Snowflake Terraform provider does not support session policies.
+
+After the Snowflake account is created, run the following script in a worksheet
+to set the appropriate session policy:
+
+```sql
+use role sysadmin;
+create database if not exists policies;
+create session policy if not exists policies.public.account_session_policy
+  session_idle_timeout_mins = 60
+  session_ui_idle_timeout_mins = 60
+;
+use role accountadmin;
+-- alter account unset session policy;  -- unset any previously existing session policy
+alter account set session policy policies.public.account_session_policy;
+```
+
+### Add IT-Ops representatives
+
+TODO: establish and document processes here.
+
+### Set up Okta SSO and SCIM
+
+TODO: establish and document processes here.
+
+## Create project git repository
+
+Create a new git repository from the CalData Infrastructure Template
+following the instructions [here](https://github.com/cagov/caldata-infrastructure-template#usage).
+
+Once you have created the repository, push it to a remote repository in GitHub.
+There are some GitHub actions that will fail because the repository is not yet
+configured to work with the new Snowflake account.
```

Comment:
I may not be able to go farther than this because the linked instructions are insufficient and I am still trying to figure out how to push the local repository to a non-existent remote repo.

Reply:
There are a lot of steps I had to take that were not listed in those docs that I'll have to add later.

Reply:
A quick tl;dr is:

- I had to create a local folder first to house all the files generated by the copier command
- That said i did not need to `cd` to that repo bc i was already in it after creating it
- I had to create a remote repo
- I had to run commands to link my local repo to that remote one
- I now have to create a PAT and tick something about workflows

Reply:
Got past this, will update the docs with the necessary info later

- links i used for reference:
   - https://github.com/orgs/community/discussions/26254
   - https://stackoverflow.com/questions/30068298/git-fatal-could-not-read-from-remote-repository-please-make-sure-you-have-th
   - https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/adding-locally-hosted-code-to-github

Reply:
> *  I had to create a local folder first to house all the files generated by the copier command

This is probably an unintended consequence of https://github.com/cagov/caldata-infrastructure-template/pull/21 . We should update the docs over there to reflect that you need to create a directory first now.

> *  That said i did not need to cd to that repo bc i was already in it after creating it

I don't understand this. After creating a directory, you are typically not placed in it, i.e., 
```bash
mkdir my-project-repo
```
does not place you in `my-project-repo`.

> * I had to create a remote repo
> * I had to run commands to link my local repo to that remote one

I'm currently thinking that instructions for how to create a GitHub repo are out-of-scope for these docs, i.e., this shouldn't cover every sub-step that needs to be done, but instead the steps that are specific to this project-set-up process.

> * I now have to create a PAT and tick something about workflows

I don't understand what you are referring here, but creating and pushing to a GitHub repo does not require a personal access token, so something seems to have gone awry.